src/allmain.c

```diff
@@ -119,13 +120,16 @@ boolean resuming;
                         mtmp->movement += mcalcmove(mtmp);
 
                     /* occasionally add another monster; since this takes
-                       place after movement has been allotted, the new
-                       monster effectively loses its first turn */
-                    if (!rn2(u.uevent.udemigod ? 25
+                       place after movement has been allotted (unless turned
+                       off in nle), the new monster effectively loses its first
+                       turn */
+                    if (nle_spawn_monsters) {
+                        if (!rn2(u.uevent.udemigod ? 25
```

Comment:
I would suggest merging these ifs and doing `if (nle_spawn_monsters && !rn2(u.uevent.udemigod ? 25` ...